network-api/networkapi/buyersguide/management/commands/aggregate_product_votes.py

```diff
@@ -0,0 +1,118 @@
+from django.core.management.base import BaseCommand
+from django.db.models import (
+    Avg,
+    IntegerField
+)
+
+from networkapi.buyersguide.models import (
+    Product,
+    RangeVote,
+    BooleanVote,
+    RangeProductVote,
+    BooleanProductVote,
+    RangeVoteBreakdown, BooleanVoteBreakdown)
+
+
+class Command(BaseCommand):
+    help = 'tally product votes and update Product records'
+
+    # define the lower and upper values for filtering and
+    # counting a product's totals in a given "bucket"
+    range_filter_values = {
+        '0': (1, 20),
+        '1': (21, 40),
+        '2': (41, 60),
+        '3': (61, 80),
+        '4': (81, 100)
+    }
+
+    def handle(self, *args, **options):
+        products = Product.objects.all()
+
+        for product in products:
+            # generate a QuerySet for this product's creepiness votes
+            creepiness_query_set = RangeVote.objects.filter(
+                attribute='creepiness',
+                product_id=product.id
+            )
+
+            # Calculate the total amount of votes for this product
+            creepiness_vote_count = creepiness_query_set.count()
+
+            # Calculate the average total creepiness score for this product, default to 50 if there aren't votes.
+            creepiness_avg = creepiness_query_set.aggregate(
+                Avg('value', output_field=IntegerField())
+            )['value__avg'] if creepiness_vote_count > 0 else 50
+
+            # define an object for recording this product's creepiness bucket totals
+            creepiness_bucket_totals = {}
+
+            # For each bucket group, filter votes on the low and high values, and record the count of each
+            for creepiness_bucket in self.range_filter_values.keys():
```

Comment:
what is `self` here?

Reply:
OH. `self` is the Test instance. it's referring to this variable: https://github.com/mozilla/foundation.mozilla.org/pull/1864/files/1f4ffa989c910c3d2cbbce049f48bc895a65e57e#diff-faeee7b1bd84dcf1f1ecd9d0661bb9b9R21

Reply:
404 on that link?

(we might want to add a comment so we keep that knowledge local to the code =D)

Reply:
yeah I rebased the branch so the hashes changed. I think it's clear if you look at parameters for the method this code is in (something I missed while skimming through the code.)